src/org/opendatakit/briefcase/ui/automation/AutomationPanelForm.java

```diff
@@ -36,17 +34,13 @@ public AutomationPanelForm() {
             .ifPresent(file -> setScriptDir(Paths.get(file.toURI()))));
   }
 
-  void onSelectScriptDir(Consumer<Path> callback) {
-    onSelectScriptDirCallbacks.add(callback);
-  }
-
-  void onGenerate(Runnable callback) {
-    generateScriptButton.addActionListener(__ -> callback.run());
+  void onGenerate(Consumer<AutomationConfiguration> callback) {
+    generateScriptButton.addActionListener(__ -> callback.accept(automationConfiguration));
```

Comment:
Remove the `automationConfiguration` member of this class and create a new `AutomationConfiguration` object inside the action listener using the current value of the UI fields.